### PR TITLE
Allow Jekyll Source Directory

### PR DESCRIPTION
**Pull request by mloberg**
Jekyll Compose assumes that the Jekyll code lives in the root directory, so if you have custom source set in your config, it won't respect that.

I've updated compose to load in configuration (adding both `--config` and `--source` options).

### CODE REVIEW

**Comment by parkr** [comment]
Seems logical to me. LGTM. 

@jekyll/compose?

File: lib/jekyll-compose/arg_parser.rb [hunk 1]
```diff
@@ -24,4 +25,8 @@ def title
   def force?
     !!options["force"]
   end
+
+  def source
+    source = config['source'].gsub(/^#{Regexp.quote(Dir.pwd)}/, '')
```
Comment (courajs) [inline lib/jekyll-compose/arg_parser.rb]: Is this variable assign necessary?

**Comment by courajs** [comment]
Great work! My only feedback would be that the specs are becoming hard to read, because of the overriding of `let`s. For example in `draft_spec.rb`, the body of [`it 'should use source directory set by config'`](https://github.com/mloberg/jekyll-compose/blob/696a74c200fcb1f502d1c4b4f3e988002d8e4e8d/spec/draft_spec.rb#L93) is exactly the same as the simple case [`it 'creates a new draft'`](https://github.com/mloberg/jekyll-compose/blob/696a74c200fcb1f502d1c4b4f3e988002d8e4e8d/spec/draft_spec.rb#L20), but it's different because `drafts_dir` is overridden down here, which causes `path` to evaluate differently at the top.

If you have the time to refactor that to make it clearer, that would be amazing! If not, this is mergable as-is.

LGTM

File: lib/jekyll-compose/arg_parser.rb (hunk 1 above)
Comment (mloberg) [inline lib/jekyll-compose/arg_parser.rb]: Other methods had an assign, so I thought I would follow that pattern. I can remove this if it isn't doing anything.

**Comment by mloberg** [comment]
@courajs I haven't done much Ruby and almost no spec testing in Ruby. Are there any guides or best practices that might help me out?

File: lib/jekyll-compose/arg_parser.rb (hunk 1 above)
Comment (courajs) [inline lib/jekyll-compose/arg_parser.rb]: Oh I hadn't noticed that. I'm pretty confident those do nothing, would you remove all three?

Comment (mloberg) [inline lib/jekyll-compose/arg_parser.rb]: Yes. I'll update that tonight.

**Comment by courajs** [comment]
I'm not sure there's a guide that would help you with this specific issue. But I think it will make things clearer to explicitly re-define `path`, rather than re-defining `drafts_path` and relying on changes to cascade. Then make it similarly explicit for `post_path` and `draft_path` in `publish_spec.rb` and `unpublish_spec.rb`.
If necessary we can always merge as-is, and I'll open a refactoring issue for someone interested to pick up.

**Comment by mloberg** [comment]
@courajs Is something like this what you're looking for?

```ruby
it 'creates draft in specified source directory' do
  site_dir = 'site'
  path = Pathname.new source_dir(site_dir, '_drafts', 'a-test-post.md')
  expect(path).not_to exist
  capture_stdout { described_class.process(args, 'source' => site_dir) }
  expect(path).to exist
end
```

File: lib/jekyll-compose/arg_parser.rb (hunk 1 above)
Comment (mloberg) [inline lib/jekyll-compose/arg_parser.rb]: I've updated this and removed all unnecessary variable assignments in that class.

**Comment by courajs** [comment]
Ah, I didn't realize the `after(:each)` hook was using draft_dir to clean up. It's seeming like a lot of effort for not too much gain, I'll just merge this. The most recent test failure is an intermittent problem that I opened an issue for (#44), but they pass locally for me.

**Pull request merged**